Bug and Fix:  Awake improvements, sanctions handling, and popup initialization  (EOSU-656)

JessTello opened the pull request:
- Changing Start to Awake fixes the nullref, due to ReportManager being null
- Added comments for better code clarity
- Renamed currentProdcutUserId to currentProductUserId for correct spelling
- Replaced ContainsKey() check with TryGetValue() for cleaner logic
- Updated conditionals to handle cases where sanctions are read but count equals zero
- Added helper method to create an entry in the sanctions list
- Fixed popup initialization to prevent null reference errors
- Corrected flow to properly send reports and receive sanctions

JessTello committed:
Bug: Fix popup initialization to prevent null reference error (EOSU-656)

- Prevents null reference exceptions during popup initialization
- Ensures Sanctions and Report feature works as intended

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -49,7 +49,7 @@ public class UIPlayerReportMenu : SampleMenuWithFriends
 
-        private void Start()
+        new private void Awake()
         {
```
You very rarely want to use the `new` function modifier. By doing this you are stopping the function in the base class from being called
```suggestion
        protected override void Awake()
        {
            base.Awake();
```

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -72,47 +72,48 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
 
```
It's not your change but can you move the call to `base.OnDestroy()` above to the bottom of the function? (this is the closest I can leave a comment 🙁). It should happen in reverse order because it's destruction

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -72,47 +72,48 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
+            ReportsManager.QueryActivePlayerSanctions(userId, (result) =>
+            {
+                QueryActivePlayerSanctionsCompleted(result, userId);
+            });
```
There's more changes here than just the nullref, it would be good to update the PR description to describe them.
I'm assuming that:
- Changing Start to Awake fixes the nullref, due to ReportManager being null
- This and the below change to use lambdas resolves the issue of `currentProdcutUserId` potentially changing in that time window

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -49,7 +49,7 @@ public class UIPlayerReportMenu : SampleMenuWithFriends
         private EOSFriendsManager FriendsManager;
```
Can you rename `currentProdcutUserId` to be spelt correctly whilst you're making nearby changes?

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -72,47 +72,48 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
             if (ReportsManager.GetCachedPlayerSanctions(out Dictionary<ProductUserId, List<Sanction>> sanctionLookup))
             {
-                if(!sanctionLookup.ContainsKey(currentProdcutUserId))
+                if (!sanctionLookup.ContainsKey(userId))
```
It's not your change but I'd generally recommend using `TryGetValue()` vs a separate `ContainsKey()` and lookup

JessTello committed:
Refactor: Improve Awake method, variable naming and sanctions handling

- Changed Awake from private to protected override and added base.Awake call
- Added comments for better code clarity
- Renamed currentProdcutUserId to currentProductUserId for correct spelling
- Replaced ContainsKey() check with TryGetValue() for cleaner logic
- Updated conditionals to handle cases where sanctions are read but count equals zero
- Added helper method to create an entry in the sanctions list

matt-clarke commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -71,80 +72,84 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
 
-                    uiEntry.TimePlaced.text = string.Empty;
-                    uiEntry.Action.text = "No Sanctions Found.";
+            if (!sanctionLookup.TryGetValue(userId, out var sanctions) || sanctions == null || sanctions.Count == 0)
```
The body of this statement and the one above are identical, the conditions can be combined into a single if statement

matt-clarke committed:
Combine if statements with identical code

EpicCliffHudson commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -71,80 +72,79 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
 
-        private void QueryActivePlayerSanctionsCompleted(Result result)
+        }
+        //Display only sanctions, reports are not visible
```
nit: add blank line after function

JessTello replied:
Implemented in the latest commit.

EpicCliffHudson commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -71,80 +72,79 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
 
-            // Update Sanctions List UI
-            if (ReportsManager.GetCachedPlayerSanctions(out Dictionary<ProductUserId, List<Sanction>> sanctionLookup))
+            if (!ReportsManager.GetCachedPlayerSanctions(out Dictionary<ProductUserId, List<Sanction>> sanctionLookup)
```
You should probably check sanctionLookup != null here like you do for sanctions.

JessTello replied:
Implemented in the latest commit.

EpicCliffHudson commented on Assets/Scripts/StandardSamples/UI/PlayerReports/UIPlayerReportMenu.cs:
```diff
@@ -71,80 +72,79 @@ public void ReportButtonOnClick(ProductUserId userId, string playerName)
-                }
+            foreach (var s in sanctions)
+            {
+                var timeText = s.TimePlaced.ToString("dd/MM/yyyy HH:mm"); 
```
Is this format specific for this instance or do we use it elsewhere?  It should probably be promoted to a named constant.

JessTello replied:
Implemented in the latest commit.

JessTello committed:
Refactor: Update PlayerReportMenu logic and date format

- Replaced hardcoded date format with constant DateFormat
- Improved null sanctionLookup checks for sanctions items

JessTello committed:
Merge branch 'release-5.2.0' into bug/eosu-656-sanctions-scene-popup-init-fix
